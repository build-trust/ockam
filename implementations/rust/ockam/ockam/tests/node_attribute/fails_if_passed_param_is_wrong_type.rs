// This test checks that #[ockam::node] causes a compile time error
// if the function is passed a param that is not of type `ockam::Context`

#[ockam::node]
async fn main(ctx: std::string::String) {}
