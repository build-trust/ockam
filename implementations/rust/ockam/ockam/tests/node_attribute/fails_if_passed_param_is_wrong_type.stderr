error: Expected `ockam::Context` found `std::string::String`
 --> $DIR/fails_if_passed_param_is_wrong_type.rs:5:20
  |
5 | async fn main(ctx: std::string::String) {}
  |                    ^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fails_if_passed_param_is_wrong_type.rs:4:1
  |
4 | / #[ockam::node]
5 | | async fn main(ctx: std::string::String) {}
  | |__________________________________________^ consider adding a `main` function to `$DIR/tests/node_attribute/fails_if_passed_param_is_wrong_type.rs`
